### PR TITLE
test: subquery test migrated from duckdb

### PR DESCRIPTION
**Pull request by CookiePieWw**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

#1168 

## What's changed and what's your intention?

As titile.

Only a few parts are migrated. Mostly because other parts are not available for greptimedb for these reasons:

1. Datafusion does not support: scalar subquery, exists, pivot ...
2. Different internal pre-defined tables, functions and types:
   - **Tables**: orders, lineitem ...
   - **Functions**: generate_series, gen_random_uuid ...
   - **Types**: array/list type
3. Different grammar: duckdb uses query context differently,
   - e.g.`SELECT (SELECT max((SELECT subq_0.c0 AS c1))) FROM (SELECT NULL AS c0) AS subq_0;`
   - Seems the correct form for greptimedb is `SELECT (SELECT max((SELECT subq_0.c0 AS c1 FROM (SELECT NULL AS c0) AS subq_0)));`
   - (Query above is still not available since datafusion does not support Scalar subqueries yet)

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by evenyag — APPROVED** [review]

File: tests/cases/standalone/common/subquery/test_neumann.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+INSERT INTO exams VALUES (1, 1, 'Database Systems', 'CS', 10, 2015);
+
+INSERT INTO exams VALUES (2, 1, 'Graphics', 'CS', 9, 2016);
+
+INSERT INTO exams VALUES (3, 2, 'Database Systems', 'CS', 7, 2015);
+
+INSERT INTO exams VALUES (4, 2, 'Graphics', 'CS', 7, 2016);
+
+SELECT s.n, e.course, e.grade FROM students s, exams e WHERE s.id=e.sid AND e.grade=(SELECT MAX(e2.grade) FROM exams e2 WHERE s.id=e2.sid) ORDER BY n, course;
```
Comment (WenyXu) [inline tests/cases/standalone/common/subquery/test_neumann.sql:17]: Add a FIXME comment

```sql
-- FIXME(cookie or ruihang?): xxx
```

Comment (waynexia) [reply tests/cases/standalone/common/subquery/test_neumann.sql:17]: File an issue and link it here

**Review by waynexia — COMMENTED** [review]
BTW, please state the original test file in duckdb, and which case(s) are skipped in *.sql file's header

**Comment by WenyXu** [comment]
@waynexia PTAL

**Comment by CookiePieWw** [comment]
Updated with sources and skipped cases. Since they distributed seperately, so I add comments for skipped cases in where they should have been.

**Pull request merged**